sdks/go/pkg/beam/testing/passert/count_test.go

```diff
@@ -22,24 +22,62 @@ import (
 	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/ptest"
 )
 
-func TestCount_Good(t *testing.T) {
-	p, s := beam.NewPipelineWithRoot()
-	col := beam.Create(s, "a", "b", "c", "d", "e")
-	count := 5
+func TestCount(t *testing.T) {
+	var tests = []struct {
+		name     string
+		elements []string
+		count    int
+	}{
+		{
+			"full",
+			[]string{"a", "b", "c", "d", "e"},
+			5,
+		},
+		{
+			"empty",
+			[]string{},
+			0,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			p, s := beam.NewPipelineWithRoot()
+			col := beam.CreateList(s, test.elements)
 
-	Count(s, col, "TestCount_Good", count)
-	if err := ptest.Run(p); err != nil {
-		t.Errorf("Pipeline failed: %v", err)
+			Count(s, col, test.name, test.count)
+			if err := ptest.Run(p); err != nil {
+				t.Errorf("Pipeline failed: %v", err)
+			}
+		})
 	}
 }
 
 func TestCount_Bad(t *testing.T) {
```

Comment:
I'd argue its worth doing since more/duplicated code => more opportunities for bugs to slip in when updates are needed and more for a future developer (maybe us) to understand ([code is a liability](https://blog.devgenius.io/code-is-a-liability-to-developers-e6ffade2b7d0)). I'm not going to block on it though, its not very important